packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts

```diff
@@ -0,0 +1,129 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import { collections } from '@salto-io/lowerdash'
+import { elements as adapterElements, config as configUtils, client as clientUtils } from '@salto-io/adapter-components'
+import { CORE_ANNOTATIONS, Element, InstanceElement, ObjectType, SaltoError } from '@salto-io/adapter-api'
+import { FilterCreator } from '../../filter'
+import { addAnnotationRecursively, findObject, setTypeDeploymentAnnotations } from '../../utils'
+import { CHUNK_SIZE, isWorkflowIdsResponse, isWorkflowResponse } from './types'
+import { DEFAULT_API_DEFINITIONS } from '../../config/api_config'
+import { JIRA_WORKFLOW_TYPE } from '../../constants'
+import JiraClient from '../../client/client'
+
+
+const { awu } = collections.asynciterable
+const { makeArray } = collections.array
+const { toBasicInstance } = adapterElements
+const { getTransformationConfigByType } = configUtils
+
+type WorkflowIdsOrFilterResult = {
+  workflowIds?: string[]
+  errors: SaltoError[]
+}
+
+const fetchWorkflowIds = async (paginator: clientUtils.Paginator): Promise<WorkflowIdsOrFilterResult> => {
+  const paginationArgs = {
+    url: '/rest/api/3/workflow/search',
+    paginationField: 'startAt',
+  }
+  const workflowValues = await awu(paginator(
+    paginationArgs,
+    page => makeArray(page.values) as clientUtils.ResponseValue[]
+  )).flat().toArray()
+  if (!isWorkflowIdsResponse(workflowValues)) {
+    return {
+      errors: [{
+        message: 'Received an invalid workflow response from service',
+        severity: 'Warning',
+      }],
+    }
+  }
+  return { workflowIds: workflowValues.map(value => value.id.entityId), errors: [] }
```

Comment:
nit - make `errors` optional here and:
```suggestion
  return { workflowIds: workflowValues.map(value => value.id.entityId) }
```